clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/CTInAppNotification.java

```diff
@@ -460,38 +462,46 @@ boolean isTablet() {
 
     void prepareForDisplay(
             FileResourceProvider fileResourceProvider,
-            final TemplatesManager templatesManager
-    ) {
+            final TemplatesManager templatesManager,
+            final StoreRegistry storeRegistry) {
+
+        final Pair<FileStore,InAppAssetsStore> storePair = new Pair<>(storeRegistry.getFilesStore(),
+                storeRegistry.getInAppAssetsStore());
+
         if (inAppType.equals(CTInAppType.CTInAppTypeCustomCodeTemplate)) {
             final List<Pair<String, CtCacheType>> fileUrlMetas = customTemplateData.getFileArgsUrls(templatesManager);
 
             int index = 0;
             while (index < fileUrlMetas.size()) {
-                Pair<String, CtCacheType> data = fileUrlMetas.get(index);
-                byte[] bytes = fileResourceProvider.fetchFile(data.getFirst());
+                Pair<String, CtCacheType> urlMeta = fileUrlMetas.get(index);
+                byte[] bytes = fileResourceProvider.fetchFile(urlMeta.getFirst());
 
-                if (bytes == null) {
+                if (bytes != null && bytes.length > 0) {
+                    FileResourcesRepoImpl.saveUrlExpiryToStore(urlMeta,storePair);
```

Comment:
FIle store and assets store can be directly used here in place to save ts. 
`FileResourcesRepoImpl` should not handle BL wrt ts.

Reply:
wanted to avoid duplicate code across classes, hence created util like function and used here.